### ISSUE TITLE
Avoid the need for @Schema decorator

### ISSUE THREAD

**Issue by felipesabino** [indigotech/graphql-schema-decorator#35]
While doing the [decorators code refactory](https://github.com/indigotech/graphql-schema-decorator/pulls?q=is%3Apr+refactor+author%3Afelipesabino+is%3Aclosed) I noticed that the GraphQL schema cold be defined on the queries/mutation/subscription directly (similar to how [`routing-controllers` project works](https://github.com/pleerock/routing-controllers#example-of-usage)) by adding the `@Query` decorator directly into query fields.

This approach goes against the current approach where everything is a `@Field` or  `@ObjectType` and some are queries defined in the schema elsewhere, making it harder to visualize what the schema actually is and IMO being less declarative than it could possibly be.

A proposed approach would be to get rid of the `@Schema` decorator in favor of loading all classes with decorators from a specific folder ([`routing-controllers` example](https://github.com/pleerock/routing-controllers#load-all-controllers-from-the-given-directory)) or passing an array of classes ([`routing-controllers` example](https://github.com/pleerock/routing-controllers#prefix-all-controllers-routes))

An example schema definition would be

```

// TypeScript definitions


// Output type
@ObjectType()
export class Foo {
    bars: string;
}

@ObjectType()
export class MyQuery {
    @Query({ type: Foo } )
    AQuery() : Promise<Foo> {
     // return Foo
    }

    @Mutation({ type: Foo } )
    AMutation(
          @Arg() aValue: string,
    ): Promise<Foo> { 
     // return Foo
    }
}

// Resulting GraphQL Schema

type Foo {
  bars: String
}

type Query {
  AQuery(): Foo
}

type Mutation {
  AMutation(aValue: String): Foo
}

schema {
  query: Query
  mutation: Mutation
}

```

This would be a breaking change but would avoid having to create a file with the `@Schema` manually referencing  all queries and mutations.

So my question is if this a reasonable (breaking) change for a 1.0.0 release?

**Comment by tibawatanabe** [comment]
I prefer this proposed approach. Though we will have a huge breaking change, it simplify things a lot.

**Comment by thiago-soliveira** [comment]
I agree. 99 out of 100 times I updated some schema, I did forget to update the correspondent `@Schema` class.

**Comment by babbarankit** [comment]
Support glob instead of folder (maybe) like TypeOrm do: 
`"entities": [
      "src/entity/**/*.ts"
   ],`

**Comment by felipesabino** [comment]
@babbarankit Good point! I had on this on my mind but forgot to add to the issue 👍

**Comment by MichalLytek** [comment]
As this had been the must-have feature for me and as I haven't seen any progress here, I've created similar project to this with "correct" organization of resolvers: 
https://19majkel94.github.io/type-graphql/docs/resolvers.html
and loading them by path (glob):
https://19majkel94.github.io/type-graphql/docs/bootstrap.html

It doesn't have all the features like pagination yet but the core is stable and extensible so all the smaller helpers will appear soon. And I would love to hear some feedback from you guys as we both started with TypeORM and routing-controllers and you are using your `graphql-schema-decorator` on production for a long time 😉
